File: state/java/PreparingState.java
package State;

// Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
class PreparingState extends VendingMachineState {
    // PreparingState durumunda ürün hazırlanması gibi işlemler yapılmaktadır.
    // Buradan sonraki durum ise ürünün verilmesi durumudur. (DeliveryState)
    @Override
    public void handleState(VendingMachine context) {
        System.out.println("PreparingState: İstenilen ürün hazırlanıyor. Lütfen bekleyiniz");

        context.setState(new DeliveryState());
    }
}

File: state/kotlin/PreparingState.kt
package State

// Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
internal class PreparingState : VendingMachineState() {
    // PreparingState durumunda ürün hazırlanması gibi işlemler yapılmaktadır.
    // Buradan sonraki durum ise ürünün verilmesi durumudur. (DeliveryState)
    override fun handleState(context: VendingMachine) {
        println("PreparingState: İstenilen ürün hazırlanıyor. Lütfen bekleyiniz")

        context.setState(DeliveryState())
    }
}